### PR TITLE
benchmark convergence time

### PR DESCRIPTION
**Pull request by mrhooray**
benchmark convergence time with child process & message passing
future work includes
- replacing interprocess messaging with swim/ringpop cli over tchannel
- worker publish checksum on membership update to independent channel instead of interprocess messaging and master/admin watch for divergence & convergence to compute convergence time; which may apply to production as well

will parameterize benchmark after #53 and other cleanup work

### CODE REVIEW

**Comment by mrhooray** [comment]
- rebased on latest master
- reintroduced leave function to ringpop; leave was refactored away, but it seems reasonable to expose leave interface for ringpop consumer
- parameterized scenario as runner argument
- added half cluster failure

```
node benchmarks/convergence-time/scenario-runner.js -h

  Usage: scenario-runner [options]

  Options:

    -h, --help          output usage information
    --scenario <value>  path to scenario relative to scenario runner
    --cycles [value]    number of cycles
    --workers [value]   number of workers
```

**Comment by charliezhang** [comment]
could you rebase and merge this?

**Comment by mrhooray** [comment]
cc @charliezhang, under default config
```
➜  ringpop git:(bench) node benchmarks/convergence-time/scenario-runner.js --scenario scenarios/half-cluster-failure.js
configuration:
- scenario scenarios/half-cluster-failure.js
- cycles 10
- workers 10
convergence time under half cluster failure
histogram data:
- count 10
- min 601
- max 1228
- mean 890.2
- median 893
- variance 35636.177777777775
- std dev 188.77546921615047
- p75 1030.25
- p95 1228
- p99 1228
```

File: index.js [hunk 1]
```diff
@@ -190,6 +192,18 @@ RingPop.prototype.setupChannel = function setupChannel() {
     createServer(this, this.channel);
 };
 
+RingPop.prototype.adminJoin = function adminJoin(callback) {
```
Comment (jwolski) [inline index.js]: Can you remove this from the RingPop prototype? We're trying to reduce the overall footprint of the "public" RingPop interface. These two shouldn't be here.

**Pull request merged**